__tests__/distributors/local.test.ts

```diff
@@ -0,0 +1,173 @@
+import * as utils from '../../src/util';
+
+import { LocalDistributor } from '../../src/distributors/local/installer';
+
+describe('Test setupJava', () => {
+  const actualJavaVersion = '11.1.10';
+  const javaPath = path.join('Java_LocalJDKFile_jdk', actualJavaVersion, 'x86');
+
+  let mockJavaBase: LocalDistributor;
+
+  let tcFind: jest.SpyInstance;
+  let coreDebug: jest.SpyInstance;
+  let coreInfo: jest.SpyInstance;
+  let coreExportVariable: jest.SpyInstance;
+  let coreAddPath: jest.SpyInstance;
+  let coreSetOutput: jest.SpyInstance;
+  let fsStat: jest.SpyInstance;
+  let spyFsReadDir: jest.SpyInstance;
+  let utilsExtractJdkFile: jest.SpyInstance;
+  let resolvePath: jest.SpyInstance;
+  let tcCacheDir: jest.SpyInstance;
+  let expectedJdkFile = 'JavaLocalJdkFile';
+
+  beforeEach(() => {
+    tcFind = jest.spyOn(tc, 'find');
+    tcFind.mockImplementation((toolname: string, javaVersion: string, architecture: string) => {
+      const semverVersion = new semver.Range(javaVersion);
+
+      if (path.basename(javaPath) !== architecture || !javaPath.includes(toolname)) {
+        return '';
+      }
+
+      return semver.satisfies(actualJavaVersion, semverVersion) ? javaPath : '';
+    });
+
+    tcCacheDir = jest.spyOn(tc, 'cacheDir');
+    tcCacheDir.mockImplementation(
+      (archivePath: string, toolcacheFolderName: string, version: string, architecture: string) =>
+        path.join(toolcacheFolderName, version, architecture)
+    );
+
+    // Spy on core methods
+
+    coreDebug = jest.spyOn(core, 'debug');
+    coreDebug.mockImplementation(() => undefined);
+
+    coreInfo = jest.spyOn(core, 'info');
+    coreInfo.mockImplementation(() => undefined);
+
+    coreAddPath = jest.spyOn(core, 'addPath');
+    coreAddPath.mockImplementation(() => undefined);
+
+    coreExportVariable = jest.spyOn(core, 'exportVariable');
+    coreExportVariable.mockImplementation(() => undefined);
+
+    coreSetOutput = jest.spyOn(core, 'setOutput');
+    coreSetOutput.mockImplementation(() => undefined);
+
+    // Spy on fs methods
+    spyFsReadDir = jest.spyOn(fs, 'readdirSync');
+    spyFsReadDir.mockImplementation(() => ['JavaTest']);
+
+    fsStat = jest.spyOn(fs, 'statSync');
+    fsStat.mockImplementation((file: string) => {
+      return { isFile: () => file === expectedJdkFile };
+    });
+
+    // Spy on util methods
+    utilsExtractJdkFile = jest.spyOn(utils, 'extractJdkFile');
+    utilsExtractJdkFile.mockImplementation(() => 'some/random/path/');
+
+    // Spy on path methods
+    resolvePath = jest.spyOn(path, 'resolve');
+    resolvePath.mockImplementation((path: string) => path);
+  });
+
+  afterEach(() => {
+    jest.resetAllMocks();
+    jest.clearAllMocks();
+    jest.restoreAllMocks();
+  });
+
+  it.each([
+    [
+      { version: '8.0.289', arch: 'x64', packageType: 'jdk' },
+      expectedJdkFile,
+      { javaVersion: '8.0.289', javaPath: path.join('Java_LocalJDKFile_jdk', '8.0.289', 'x64') }
```

Comment:
Looks like `javaPath` uses the same value from input object. Can we build it right in test case to avoid code duplication?

Reply:
Done. Thank you.